Sources/Afterpay/AfterpayV3.swift

```diff
@@ -0,0 +1,258 @@
+  let shopDirectoryMerchantId: String
+  let region: Region
+  let environment: Environment
+
+  /// Creates a collection of options and values required to interact with the Afterpay API.
+  /// - Parameters:
+  ///   - shopDirectoryMerchantId: A unique merchant identifier.
+  ///   - region: The region serviced by the merchant.
+  ///   - environment: The environment. Use `sandbox` for development purposes.
+  public init(
+    shopDirectoryMerchantId: String,
+    region: Region,
+    environment: Environment
+  ) {
+    self.shopDirectoryMerchantId = shopDirectoryMerchantId
+    self.region = region
+    self.environment = environment
+  }
+
+  // MARK: - Computed properties
+
+  var shopDirectoryId: String {
+    switch (region, environment) {
+    case (_, .sandbox):
+      return "cd6b7914412b407d80aaf81d855d1105"
+    case (_, .production):
+      return "e1e5632bebe64cee8e5daff8588e8f2f05ca4ed6ac524c76824c04e09033badc"
+    }
+  }
+
+  var v3BaseUrl: URL {
+    switch (region, environment) {
+    case (.US, .sandbox):
+      return URL(string: "https://api-plus.us-sandbox.afterpay.com/v3/button")!
+    case (.US, .production):
+      return URL(string: "https://api-plus.us.afterpay.com/v3/button")!
+    // Currently the same URL as the US region
+    case (.CA, .sandbox):
+      return URL(string: "https://api-plus.us-sandbox.afterpay.com/v3/button")!
+    case (.CA, .production):
+      return URL(string: "https://api-plus.us.afterpay.com/v3/button")!
+    }
+  }
+
+  var v3CheckoutUrl: URL {
+    self.v3BaseUrl
+  }
+
+  var v3CheckoutConfirmationUrl: URL {
+    v3BaseUrl.appendingPathComponent("confirm")
+  }
+
+  var v3ConfigurationUrl: URL {
+    var components = URLComponents(
+      url: v3BaseUrl.appendingPathComponent("merchant/config"),
+      resolvingAgainstBaseURL: false
+    )
+    components?.queryItems = [
+      URLQueryItem(name: "shopDirectoryId", value: shopDirectoryId),
+      URLQueryItem(name: "shopDirectoryMerchantId", value: shopDirectoryMerchantId),
+    ]
+    guard let url = components?.url else {
+      fatalError("Could not create valid URL for `\(Self.self).v3ConfigurationUrl`")
+    }
+    return url
+  }
+
+  // MARK: - Inner type
+
+  /// Regions supporting V3 checkouts
+  public enum Region {
+    case US
+    case CA
+
+    var locale: Locale {
+      switch self {
+      case .US: return Locales.enUS
+      case .CA: return Locales.enCA
+      }
+    }
+
+    var currencyCode: String {
+      guard let currencyCode = self.locale.currencyCode else {
+        switch self {
+        case .US: return "USD"
+        case .CA: return "CAD"
+        }
+      }
+      return currencyCode
+    }
+
+    private static var formatter: NumberFormatter = {
+      var formatter = NumberFormatter()
+      formatter.numberStyle = .decimal
+      // ISO 4217 specifies 2 decimal points
+      formatter.maximumFractionDigits = 2
+      formatter.roundingMode = .halfEven // Banker's rounding
```

Comment:
"Banker's rounding"? What is this formatter used for? The checkout and payment scheduler do not use this rounding method.